src/ring_theory/ideal/local_ring.lean

```diff
@@ -32,53 +32,43 @@ universes u v w
 
 /-- A commutative ring is local if it has a unique maximal ideal. Note that
   `local_ring` is a predicate. -/
-class local_ring (R : Type u) [comm_ring R] extends nontrivial R : Prop :=
-(is_local : ∀ (a : R), (is_unit a) ∨ (is_unit (1 - a)))
+class local_ring (R : Type u) [comm_semiring R] extends nontrivial R : Prop :=
```

Comment:
Can you add back the old `nonunits_add`? Now it is immediate from the definition, but it could be useful.

Reply:
The old `nonunits_add` can be accessed with the same name `local_ring.nonunits_add` in the new definition. I would like to remark that the proof of the old `nonunits_add` is moved to the proof of the lemma `local_of_is_unit_or_is_unit_one_sub_self`, which plays the same role as the old version constructor of `local_ring`.

Reply:
But since `local_ring` is a class in practice there will be no `h : local_ring R`, so it will b annoying to access it. Or do you see another way?

Reply:
Yes, if I understand correctly, the current [local_ring.nonunits_add](https://leanprover-community.github.io/mathlib_docs/ring_theory/ideal/local_ring.html#local_ring.nonunits_add) is intended to be used either with the fully specified name `local_ring.nonunit_add` or with `nonunits.add` followed by `open local_ring` (or in the namespace `local_ring`). With the new definition this lemma is exactly the projection of `local_ring` that is automatically generated when the class is defined.

Reply:
Ah yes, it works because the lemma I want is automatically created by Lean.